pkg/agent/multicast/mcast_route_linux.go

```diff
@@ -71,10 +77,82 @@ func (c *MRouteClient) run(stopCh <-chan struct{}) {
+	// multicast routes' statistics are getting updated by this process as well.
+	go wait.NonSlidingUntil(c.updateMrouteStats, time.Minute, stopCh)
+
 	for i := 0; i < int(workerCount); i++ {
 		go c.worker(stopCh)
 	}
 	<-stopCh
 	c.socket.FlushMRoute()
 	syscall.Close(c.socket.GetFD())
 }
+
+func (c *MRouteClient) updateMulticastRouteStatsEntry(entry *multicastRouteEntry) (isDeleted bool, newEntry *multicastRouteEntry) {
+	packetCount, err := c.socket.GetMroutePacketCount(net.ParseIP(entry.src), net.ParseIP(entry.group))
+	if err != nil {
+		klog.ErrorS(err, "Failed to get packet count for outbound multicast route", "outboundRoute", entry)
+		return false, nil
+	}
+	packetCountDiff := packetCount - entry.pktCount
+	now := time.Now()
+	klog.V(4).Infof("Multicast route %v routes %d packets in last %s", entry, packetCountDiff, mRouteTimeout.String())
+	if packetCountDiff == uint32(0) && now.Sub(entry.updatedTime) > mRouteTimeout {
+		return true, nil
+	}
+	newEntry = &multicastRouteEntry{group: entry.group, src: entry.src, pktCount: packetCount, updatedTime: now}
+	return false, newEntry
+}
+
+func (c *MRouteClient) updateInboundMrouteStats() {
+	deletedInboundRoutes := make([]*inboundMulticastRouteEntry, 0)
+	for _, obj := range c.inboundRouteCache.List() {
+		entry := obj.(*inboundMulticastRouteEntry)
+		isDeleted, newEntry := c.updateMulticastRouteStatsEntry(&entry.multicastRouteEntry)
+		if isDeleted {
+			deletedInboundRoutes = append(deletedInboundRoutes, entry)
+		} else if newEntry != nil {
+			newInboundEntry := inboundMulticastRouteEntry{*newEntry, entry.vif}
+			c.inboundRouteCache.Update(&newInboundEntry)
+		}
+	}
+	for _, inboundRoute := range deletedInboundRoutes {
+		klog.V(2).InfoS("Deleting staled inbound multicast route", "group", inboundRoute.group, "source", inboundRoute.src)
+		err := c.deleteInboundMRoute(inboundRoute)
+		if err != nil {
+			klog.ErrorS(err, "Failed to delete inbound multicast route", "group", inboundRoute.group, "source", inboundRoute.src, "VIF", inboundRoute.vif)
+			continue
+		}
+	}
+}
+
+func (c *MRouteClient) updateOutboundMrouteStats() {
+	deletedOutboundRoutes := make([]*outboundMulticastRouteEntry, 0)
+	for _, obj := range c.outboundRouteCache.List() {
+		entry := obj.(*outboundMulticastRouteEntry)
+		isDeleted, newEntry := c.updateMulticastRouteStatsEntry(&entry.multicastRouteEntry)
+		if isDeleted {
+			deletedOutboundRoutes = append(deletedOutboundRoutes, entry)
+		} else if newEntry != nil {
+			newOutboundEntry := outboundMulticastRouteEntry{*newEntry}
+			c.outboundRouteCache.Update(&newOutboundEntry)
+		}
+	}
+	for _, outboundRoute := range deletedOutboundRoutes {
+		klog.V(2).InfoS("Deleting staled outbound multicast route", "group", outboundRoute.group, "source", outboundRoute.src)
+		err := c.deleteOutboundMRoute(outboundRoute)
+		if err != nil {
+			klog.ErrorS(err, "Failed to delete outbound multicast route", "group", outboundRoute.group, "source", outboundRoute.src)
+			continue
```

Comment:
Seems the `continue` is useless, it continues even if you don't use an explicit command.